### PR TITLE
Adding .run() method to interpreter

### PR DESCRIPTION
**Pull request by gabewillen**
Adding .run() method to interpreter to keep the process alive.

### CODE REVIEW

**Comment by gabewillen** [comment]
I'm looking into how I can add a test to determine that this is actually keeping the process alive.  If anyone has an idea let me know. Also thinking about adding a `.onIdle(...)` method similar to `.onDone()` that is called when the interpreter is idling.

File: src/interpreter.ts [hunk 1]
```diff
@@ -388,6 +388,18 @@ export class Interpreter<
...
+
+  /**
+   * Starts the interpreter keeping the current process running
+   * @param initialState The state to start the statechart from
+   */
+  public run(initialState?: State<TContext, TEvent> | StateValue): Interpreter<TContext, TStateSchema, TEvent> {
+      if (!this.initialized) {
+        this.start(initialState);
+      }
+      setImmediate(this.idle.bind(this));
```
Comment (davidkpiano) [inline src/interpreter.ts]: This won't work for browser environments, by the way

Comment (gabewillen) [reply src/interpreter.ts]: Updated this to use setTimeout(..., 0) instead.

**Comment by davidkpiano** [comment]
I think it would be better to provide guidance on how to do this in the docs rather than baking it into the interpreter, for now. Thank you, though!

**Pull request closed without merging**